controlpanel/api/models/user.py

```diff
@@ -83,10 +83,28 @@ def reset_mfa(self):
         auth0.ManagementAPI().reset_mfa(self.auth0_id)
 
     def save(self, *args, **kwargs):
+        request_user = None
+        request = CrequestMiddleware.get_request()
+        if request:
+            request_user = request.user
+
         try:
-            User.objects.get(pk=self.pk)
+            existing = User.objects.get(pk=self.pk)
+
         except User.DoesNotExist:
             cluster.User(self).create()
+            if self.is_superuser:
+                slack.notify_team(
+                    slack.CREATE_SUPERUSER_MESSAGE.format(username=self.username),
+                    request_user=request_user,
```

Comment:
Makes more sense to change to `username=request_user.username` IMHO as the `slack.notify_team()` doesn't really need the coupling and dependency with the `User` model (or whatever type it's the request user).